src/variables/body/variablesSearch.tsx

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { ReactWidget } from '@jupyterlab/apputils';
+
+import { Widget, PanelLayout } from '@phosphor/widgets';
+
+import React, { useState, useRef } from 'react';
+
+import useOutsideClick from './useOutsideClick';
+
+import { Variables } from '../index';
+
+export class VariablesSearch extends Widget {
+  constructor(model: any) {
+    super();
+    this.addClass('jp-DebuggerVariables-Search');
+
+    const layout = new PanelLayout();
+    this.layout = layout;
+    this.node.style.overflow = 'visible';
+    this.scope = new VariableScopeSearch();
+    this.search = new VariableSearchInput(model);
+
+    layout.addWidget(this.scope);
+    layout.addWidget(this.search);
+  }
+
+  readonly scope: Widget;
+  readonly search: Widget;
+}
+
+const SearchComponent = ({ model }: any) => {
+  const [state, setState] = useState('');
+  model.filter = state;
+  return (
+    <input
+      placeholder="Search..."
+      value={state}
+      onChange={e => {
+        setState(e.target.value);
+      }}
+    />
+  );
+};
+
+class VariableSearchInput extends ReactWidget {
+  search: string;
+  model: Variables.IModel;
+  constructor(model: Variables.IModel) {
+    super();
+    this.model = model;
+    this.search = model.filter;
+    this.node.style;
+    this.addClass('jp-SearchInput');
+  }
+
+  render() {
+    return <SearchComponent model={this.model} />;
+  }
+}
+
+const VariablesMenu = ({ config }: any) => {
+  const [toggleState, setToggle] = useState(false);
+  const [scope, setScope] = useState('local');
+  const wrapperRef = useRef(null);
+
+  const onClickOutSide = () => {
+    setToggle(false);
+  };
+
+  const toggle = (e: any) => {
+    setToggle(!toggleState);
+  };
+
+  useOutsideClick(wrapperRef, onClickOutSide);
+
+  const changeScope = (newScope: string) => {
+    if (newScope === scope) {
+      return;
+    }
+    setScope(newScope);
+    setToggle(false);
+  };
+
+  const List = (
+    <ul className="jp-MenuComponent">
```

Comment:
`jp-MenuComponent` name is too generic. You might not even need it, though. You might be able to do:
`.jp-DebuggerVariables ul`